### ISSUE TITLE
With latest Bot V4 and Asp.Net Core 2.2 getting bot auth error

### ISSUE THREAD

**Issue by vikramdadwal** [MicrosoftDocs/bot-docs#1159]
Method not found: 'Void Microsoft.IdentityModel.Clients.ActiveDirectory.AuthenticationContext..ctor(System.String, Boolean, Microsoft.IdentityModel.Clients.ActiveDirectory.TokenCache, System.Net.Http.HttpClient)'.
   
   at Microsoft.Bot.Connector.Authentication.AdalAuthenticator..ctor(ClientCredential clientCredential, OAuthConfiguration configurationOAuth, HttpClient customHttpClient)
   at Microsoft.Bot.Connector.Authentication.MicrosoftAppCredentials.&lt;&gt;c__DisplayClass8_0.&lt;.ctor&gt;b__0() in d:\a\1\s\libraries\Microsoft.Bot.Connector\Authentication\MicrosoftAppCredentials.cs:line 96
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
--- End of stack trace from previous location where exception was thrown ---
   at System.Lazy`1.CreateValue()
   at Microsoft.Bot.Connector.Authentication.MicrosoftAppCredentials.GetTokenAsync(Boolean forceRefresh) in d:\a\1\s\libraries\Microsoft.Bot.Connector\Authentication\MicrosoftAppCredentials.cs:line 223
   at Microsoft.Bot.Connector.Authentication.MicrosoftAppCredentials.ProcessHttpRequestAsync(HttpRequestMessage request, CancellationToken cancellationToken) in d:\a\1\s\libraries\Microsoft.Bot.Connector\Authentication\MicrosoftAppCredentials.cs:line 207
   at Microsoft.Bot.Connector.Authentication.MicrosoftAppCredentials.ProcessHttpRequestAsync(HttpRequestMessage request, CancellationToken cancellationToken) in d:\a\1\s\libraries\Microsoft.Bot.Connector\Authentication\MicrosoftAppCredentials.cs:line 207
   at Microsoft.Bot.Connector.Conversations.ReplyToActivityWithHttpMessagesAsync(String conversationId, String activityId, Activity activity, Dictionary`2 customHeaders, CancellationToken cancellationToken) in d:\a\1\s\libraries\Microsoft.Bot.Connector\Conversations.cs:line 1176
   at Microsoft.Bot.Connector.ConversationsExtensions.ReplyToActivityAsync(IConversations operations, String conversationId, String activityId, Activity activity, CancellationToken cancellationToken) in d:\a\1\s\libraries\Microsoft.Bot.Connector\ConversationsExtensions.cs:line 241
   at Microsoft.Bot.Connector.ConversationsExtensions.ReplyToActivityAsync(IConversations operations, String conversationId, String activityId, Activity activity, CancellationToken cancellationToken) in d:\a\1\s\libraries\Microsoft.Bot.Connector\ConversationsExtensions.cs:line 241
   at Microsoft.Bot.Builder.BotFrameworkAdapter.SendActivitiesAsync(ITurnContext turnContext, Activity[] activities, CancellationToken cancellationToken) in d:\a\1\s\libraries\Microsoft.Bot.Builder\BotFrameworkAdapter.cs:line 327
   at Microsoft.Bot.Builder.BotFrameworkAdapter.SendActivitiesAsync(ITurnContext turnContext, Activity[] activities, CancellationToken cancellationToken) in d:\a\1\s\libraries\Microsoft.Bot.Builder\BotFrameworkAdapter.cs:line 327
   at Microsoft.Bot.Builder.TurnContext.&lt;&gt;c__DisplayClass22_0.&lt;&lt;SendActivitiesAsync&gt;g__SendActivitiesThroughAdapter|1&gt;d.MoveNext() in d:\a\1\s\libraries\Microsoft.Bot.Builder\TurnContext.cs:line 267
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Bot.Builder.TurnContext.&lt;&gt;c__DisplayClass22_0.&lt;&lt;SendActivitiesAsync&gt;g__SendActivitiesThroughAdapter|1&gt;d.MoveNext() in d:\a\1\s\libraries\Microsoft.Bot.Builder\TurnContext.cs:line 267
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Bot.Builder.TurnContext.SendActivityAsync(IActivity activity, CancellationToken cancellationToken) in d:\a\1\s\libraries\Microsoft.Bot.Builder\TurnContext.cs:line 196
   at Microsoft.Bot.Builder.TurnContext.SendActivityAsync(IActivity activity, CancellationToken cancellationToken) in d:\a\1\s\libraries\Microsoft.Bot.Builder\TurnContext.cs:line 196
   

Need quick help please

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: bc8a0643-f219-a48f-666e-95bc9f0f351c
* Version Independent ID: 81578919-249c-b3e7-b367-183d70135890
* Content: [Troubleshooting Bot Framework Authentication - Bot Service](https://docs.microsoft.com/en-us/azure/bot-service/bot-service-troubleshoot-authentication-problems?view=azure-bot-service-4.0)
* Content Source: [articles/bot-service-troubleshoot-authentication-problems.md](https://github.com/MicrosoftDocs/bot-docs/blob/live/articles/bot-service-troubleshoot-authentication-problems.md)
* Service: **bot-service**
* GitHub Login: @DeniseMak
* Microsoft Alias: **v-demak**

**Comment by DanDev33** [comment]
Change your Asp.Net version to 2.1.1 and see if this resolves your issue.

**Comment by Eguthrie3214** [comment]
I'm encountering this issue as well. I'm on 2.1.1.

**Comment by Eguthrie3214** [comment]
Found the issue. 
Package <PackageReference Include="Microsoft.IdentityModel.Clients.ActiveDirectory" Version="5.0.5" />
Is overriding some functionality and failing.

**Comment by AAATechGuy** [comment]
> Found the issue.

@Eguthrie3214 / @vikramdadwal  so, how did you fix it? By removing this package? I use this ActiveDirectory package to authenticate AAD in SQL. I cannot remove it. What's the alternative?

**Comment by Eguthrie3214** [comment]
I removed the package. You could maybe try a different version. The optimal solution would be to update the bot package to use the most current aad functionality but I’m not sure how to accomplish that.

**Comment by AAATechGuy** [comment]
Ok. So, here's what I found... I compared with an earlier checked-in version. 

![image](https://user-images.githubusercontent.com/3484074/59554998-b94d6400-8f60-11e9-939d-ccb3a867ecab.png)

After upgrade to the following, it started throwing exceptions.

![image](https://user-images.githubusercontent.com/3484074/59555001-cf5b2480-8f60-11e9-9c52-637c307f5252.png)

Asp.Net Core team should really fix version upper bound, if they are not supported.
